Helper/Data.php

```diff
@@ -215,59 +225,49 @@ public function subscribe($email, $data = [], $update = 0)
     }
 
     /**
-     * Get Subscribe/Import Customer to Smaily by email with AutoResponder ID
+     * Get Subscribe/Import Customer to Smaily by email with OPT-IN trigger.
      *
      * @return array
      *  Smaily api response
      */
-    public function subscribeAutoresponder($aid, $email, $data = [])
+    public function optInSubscriber($email, $data = [])
     {
         $address = [
             'email' => $email,
         ];
 
         if (!empty($data)) {
-            $fields = explode(',', $this->getGeneralConfig('fields'));
             foreach ($data as $field => $val) {
-                if ($field === 'name' || in_array($field, $fields, true)) {
                     $address[$field] = trim($val);
-                }
             }
         }
 
         $post = [
-            'autoresponder' => $aid,
             'addresses' => [$address],
         ];
 
         return $this->callApi('autoresponder', $post, 'POST');
     }
 
     /**
-     * Send newsletter subscribers to Smaily
+     * Send newsletter subscribers to Smaily.
      *
-     * @return array
-     *  Smaily api response
+     * @param array $list Subscribers list in batches.
+     * @return boolean Success/Failure status
      */
     public function cronSubscribeAll($list)
     {
-        $data = [];
-        // Get unsubscribers from Smaily
-        $unsubscribers = $this->getUnsubscribers();
-        // Populate unsubscribers emails array
-        $unsubscribers_emails= [];
-        foreach ($unsubscribers as $unsubscriber) {
-            if (isset($unsubscriber['email'])) {
-                $unsubscribers_emails[] = $unsubscriber['email'];
-            }
-        }
-        // Update only subscribers who are still subscribed
-        foreach ($list as $row) {
-            if (!in_array($row['email'], $unsubscribers_emails)) {
-                $data[] = $row;
+        $success = true;
+
+        foreach ($list as $batch) {
+            $response = $this->callApi('contact', $batch, 'POST');
+            if (!array_key_exists('message', $response) ||
+                array_key_exists('message', $response) && $response['message'] !== 'OK') {
+                    $success = false;
```

Comment:
Currently I implemented it like that. Should I cancel sending rest of the batches when one fails? I was thinking if you have a really big customer list like 200 000 customers and one batch fails, should it cancel rest of the update too?

Reply:
I think it should stop submitting rest of the data, and it shouldn't update last synchronization time, because otherwise all of the data in one batch will be never synced and customer data goes out of sync.